backend/worker.go

```diff
@@ -90,23 +90,20 @@ func (w *worker) Start(ctx context.Context) {
 	// TODO: Check for already started worker
 	ctx, cancel := context.WithCancel(ctx)
 	w.cancel = cancel
-
+	b := &backoff.ExponentialBackOff{
+		InitialInterval:     50 * time.Millisecond,
+		MaxInterval:         5 * time.Second,
+		Multiplier:          1.05,
+		RandomizationFactor: 0.05,
+		Stop:                backoff.Stop,
+		Clock:               backoff.SystemClock,
+	}
 	go func() {
+		ticker := backoff.NewTicker(b)
+		defer ticker.Stop()
 	loop:
 		for {
-			b := &backoff.ExponentialBackOff{
-				InitialInterval:     50 * time.Millisecond,
-				MaxInterval:         5 * time.Second,
-				Multiplier:          1.05,
-				RandomizationFactor: 0.05,
-				Stop:                backoff.Stop,
-				Clock:               backoff.SystemClock,
-			}
 			b.Reset()
```

Comment:
It looks like we're resetting the exponential backoff object every iteration of the loop now. What is the implication of this? Doesn't this cause us to only use the initial interval, thus defeating the purpose of using the ExponentialBackoff library?